xcdat/axis.py

```diff
@@ -0,0 +1,102 @@
+from typing import Literal
+
+import numpy as np
+import xarray as xr
+
+Axis = Literal["lat", "lon"]
+
+
+@xr.register_dataset_accessor("axis")
+class AxisAccessor:
+    axes_map = {
+        "lat": {
+            "coords": ["lat", "latitude"],
+            "bounds_vars": ["lat_bnds", "latitude_bnds"],
+        },
+        "lon": {
+            "coords": ["lon", "longitude"],
+            "bounds_vars": ["lon_bnds", "longitude_bnds"],
+        },
+    }
+
+    def __init__(self, xarray_obj: xr.Dataset):
+        self._obj = xarray_obj
+
+    def get_bounds(self, axis: Axis) -> xr.DataArray:
+        """Get bounds for an axis.
+
+        :param axis: "lat" or "lon" axis
+        :type axis: Axis
+        :return: The axis bounds, either existing bounds or calculated bounds
+        :rtype: xr.DataArray
+        """
+        self._obj["calculated_bounds"] = False
+        bounds_vars = AxisAccessor.axes_map[axis]["bounds_vars"]
+        matching_bounds_var = None
+
+        for var in bounds_vars:
+            matching_bounds_var = self._obj.data_vars.get(var)
+            if matching_bounds_var is not None:
+                break
+
+        if matching_bounds_var is None:
+            return self._calc_bounds(axis)
+
+        return matching_bounds_var
+
+    def _calc_bounds(self, axis: Axis) -> xr.DataArray:
```

Comment:
This method was based on https://github.com/lee1043/XCDAT/blob/master/compare_cdat_xarray/explore_spatial_averages.ipynb
(`get_lat_bounds`)

Reply:
I think this will not work correctly for longitude (since longitude limits are not +/- 90). I also think the algorithm will need to be aware of whether the longitude limits are [0:360] or [-180:180]. 

The method in the example above was meant to be pretty minimal (and not necessarily robust). I think we probably need to look at [getBounds](https://github.com/CDAT/cdms/blob/master/Lib/axis.py#L2459), which is more sophisticated.

Reply:
Thanks, after doing a call stack trace starting from `getBounds`, it leads to [`genGenericBounds`](https://github.com/CDAT/cdms/blob/dd41a8dd3b5bac10a4bfdf6e56f6465e11efc51d/Lib/axis.py#L2171) which contains the core logic to handle both axis.